packages/instant/src/util/heap.ts

```diff
@@ -0,0 +1,85 @@
+import { ObjectMap } from '@0x/types';
+
+import { HEAP_ANALYTICS_DEVELOPMENT_APP_ID } from '../constants';
+
+export interface HeapAnalytics {
+    loaded: boolean;
+    identify(id: string, idType: string): void;
+    track(eventName: string, eventProperties?: ObjectMap<string | number>): void;
+    resetIdentity(): void;
+    addUserProperties(properties: ObjectMap<string | number>): void;
+    addEventProperties(properties: ObjectMap<string | number>): void;
+    removeEventProperty(property: string): void;
+    clearEventProperties(): void;
+}
+interface ModifiedWindow {
+    heap?: HeapAnalytics;
+    zeroExInstantLoadedHeap?: boolean;
+}
+const getWindow = (): ModifiedWindow => {
+    return window as ModifiedWindow;
+};
+
+const setupZeroExInstantHeap = () => {
+    const curWindow = getWindow();
+    // Set property to specify that this is zeroEx's heap
+    curWindow.zeroExInstantLoadedHeap = true;
+
+    // Typescript-compatible version of https://docs.heapanalytics.com/docs/installation
+    /* tslint:disable */
+    ((window as any).heap = (window as any).heap || []),
+        ((window as any).heap.load = function(e: any, t: any) {
+            ((window as any).heap.appid = e), ((window as any).heap.config = t = t || {});
+            var r = t.forceSSL || 'https:' === (document.location as Location).protocol,
```

Comment:
do we need to be using var's here?

Reply:
This is just a copy-paste of heap's integration code (https://docs.heapanalytics.com/docs/installation) with small mods to make TS compile it